examples/transpiler/babel.bzl

```diff
@@ -1,32 +1,37 @@
 "Adapter from the Babel CLI to the ts_project#transpiler interface"
 
+load("@aspect_rules_js//js:defs.bzl", "js_library")
```

Comment:
since this PR claims to fix an issue, I think you should add another `ts_project` call to this example that demonstrates the issue is fixed, like a `filegroup` in the `srcs` that goes through this same `babel` rule and assert on the output files (see the new testing.bzl#assert_outputs in bazel-lib)

Reply:
Note that currently this PR keeps this babel example as-is... one babel invocation per .ts file, and currently it `fail`s for non-ts srcs such as `filegroup`s.

We should have a proper example of what is being fixed and maybe this is the place to do it, or could demo both here...

Reply:
yes that's true - we wanted to avoid a cyclic dependency rules_ts <-> rules_swc which is why it's like this. I think you could add that test in the mock_transpiler instead.

Reply:
I think a proper test needs a rule and not just a macro? So the rule can expand the target (such as a `filegroup`) into individual files to declare the outputs. Or output a directory...? 🤔

Reply:
Yeah either a tiny rule or a directory makes sense

Reply:
It looks like a directory isn't possible because the `*_typings` rule outputs the `.d.ts` into that directory, so the transpiler can't declare the full directory as an output. That is (and was) a restriction that I'm not sure about... 🤔 

Any ideas for the easiest way to wrap babel in a rule?

Reply:
Can't you use --outDir to put babel output in a subfolder?

Reply:
We could, but then it's in a different directory then the `.d.ts` outputted by the typechecking?

Note that this is putting it in a subfolder, but both the .js and .d.ts are put in the same subfolder atm.

Reply:
the types and js files don't have to be in the same folder, package.json fields allow an npm package to ship with them that way. We just need something minimal here to exercise the new logic. If you want to make some other rule to exercise it in a way that makes more sense to users, I think that's okay

Reply:
See the changes to ts/test/mock_transpiler.bzl, similar to the change in rules_swc